city.lua

```diff
@@ -721,6 +721,8 @@ local function addBuildingLocations(city, recentCoordinates, allowCenter)
     if recentCoordinates ~= nil then
         local surrounds = getSurroundingCoordinates(recentCoordinates, 1, false, allowCenter)
         for _, value in ipairs(surrounds) do
+            -- TODO: this will probably skip corners when called from on_entity_destroyed event
+            -- are edges supposed to be roads or what?
```

Comment:
The reason for this check lies within the self-expanding city grid. During road expansion we may detect that we've reached the edge of the grid: https://github.com/bahrmichael/factorio-tycoon/blob/888831dee6f2f586b82de6013882f281a5f40fab/city.lua#L972-L986

When that happens, it adds a new row/column at each side of the grid to give more space to grow into.

The check in this file just says "if the grid hasn't gone far enough yet, then ignore that section and try again in the future". This leads to the city having a higher chance to fill out the whole grid, instead of growing along one tree.